### PR TITLE
Render empty GroupedList headers

### PR DESCRIPTION
**Pull request by elliottsj**
#### Pull request checklist

- [ ] Addresses an existing issue
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Enable empty group headers (i.e. groups where `group.count === 0`) to be rendered. For example:

capture

Currently, I think this would be a breaking change since it changes the behavior of GroupedList, so we'd probably need some kind of option to enable rendering of empty group headers. I can think of a couple ways:

1. Add a boolean prop to `GroupedList` and `DetailsList`, e.g. `showEmptyGroups`. Default: `false`
2. Add a boolean property to the `IGroup` interface, e.g. `showIfEmpty`. This way, users can choose to render some empty groups, but not others.

### CODE REVIEW

**Comment by betrue-final-final** [comment]
Yes, I think this would be a good option, as long as it's not the default.

**Comment by micahgodbolt** [comment]
we usually like to have booleans that default to true, so maybe suppressEmptyGroups

**Comment by elliottsj** [comment]
I've now added the `showEmptyGroups` prop, as discussed on Teams.

File: packages/office-ui-fabric-react/src/utilities/selection/Selection.ts [hunk 1]
```diff
@@ -172,6 +172,10 @@ export class Selection implements ISelection {
   }
 
   public isRangeSelected(fromIndex: number, count: number): boolean {
+    if (count === 0) {
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/utilities/selection/Selection.ts]: Why is this change necessary?

Comment (elliottsj) [reply packages/office-ui-fabric-react/src/utilities/selection/Selection.ts]: This prevents an empty group header from appearing "selected". This is the call site: https://github.com/OfficeDev/office-ui-fabric-react/blob/e9306660b0538cc4c30e6d485bbf575a7c78c28f/packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx#L129

Comment (elliottsj) [reply packages/office-ui-fabric-react/src/utilities/selection/Selection.ts]: It looks like `isRangeSelected` is only ever called directly by GroupedListSection, and indirectly by GroupedList via `toggleRangeSelected`. I've done some manual testing of selecting & deselecting groups, so I think this is a safe change.
